op-mode-definitions/reboot.xml

```diff
@@ -4,35 +4,43 @@
     <properties>
       <help>Reboot the system</help>
     </properties>
-    <command>/opt/vyatta/bin/sudo-users/vyatta-reboot.pl --action reboot</command>
+    <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --reboot now</command>
+
     <children>
       <leafNode name="now">
         <properties>
-          <help>Reboot the system without confirmation</help>
+          <help>Reboot the system imidiently</help>
         </properties>
-        <command>/opt/vyatta/bin/sudo-users/vyatta-reboot.pl --action reboot --now</command>
+        <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --yes --reboot now</command>
       </leafNode>
-      
+
       <leafNode name="cancel">
         <properties>
           <help>Cancel a pending reboot</help>
         </properties>
-        <command>/opt/vyatta/bin/sudo-users/vyatta-reboot.pl --action reboot_cancel</command>
+        <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --cancel</command>
       </leafNode>
 
+    <tagNode name="in">
+      <properties>
+        <help>Reboot in X minutes</help>
+        <completionHelp>
+          <list>Minutes</list>
+        </completionHelp>
+      </properties>
+      <command>sudo ${vyos_op_scripts_dir}/powerctrl.py --yes --reboot $3</command>
+
+    </tagNode>
 
     <tagNode name="at">
       <properties>
         <help>Reboot at a specific time</help>
         <completionHelp>
           <list>HH:MM</list>
-          <list>MMDDYY</list>
```

Comment:
See comments in poweroff command

Reply:
see comment on poweroff.xml

Reply:
implemented as reboot/poweroff at HH:MM date DD.MM.YYYY. the difference from 1.1.8 is that a time needs to be configured. but can still be sett to 00:00 to give same action as in 1.1.8.. The current implementation uses systemd-shutdownd's "add by minutes" functionality, and so the second counter is ticking and i cant reset it to :00 
Adding only time with the format HH:MM uses "add by timespec" inside systemd-shutdownd and automatically resets seconds counter to :00